frontend/src/components/RecipeDraft.js

```diff
@@ -0,0 +1,48 @@
+import {makeStyles, TextField} from "@material-ui/core";
+import {useState} from "react";
+import IngredientEditor from "./IngredientEditor";
+
+export default function RecipeDraft({recipe, onRecipeUpdated}) {
+    const [ingredients, setIngredients] = useState(recipe.ingredients);
+    const classes = useStyles();
+
+    const onNameChange = (changeEvent) => {
+        recipe.name = changeEvent.value;
+        onRecipeUpdated(recipe);
+    }
+
+    const onIngredientChange = (updatedIngredient) => {
+        setIngredients(ingredients.map(ingredient =>
+            ingredient.id === updatedIngredient.id ? updatedIngredient : ingredient))
+        recipe.ingredients =ingredients
+        onRecipeUpdated(recipe)
```

Comment:
This looks strange. You modify the incoming `recipe` object within this child component. To prevent side-effects within the parent component, you may spread it `onRecipeUpdate({...recipe, ingredients})` before? Or am i missing something?

Reply:
I'll think about it when it comes up in later tasks, at the moment I am not quite sure how to approach this without breaking it.

Reply:
🎉 Sounds familar to me. :ninja: code!